app/routes/articles/components/ArticleDetail.tsx

```diff
@@ -1,53 +1,129 @@
+import { useAppDispatch, useAppSelector, useHelmet } from 'app/store/hooks';
+import type { PublicArticle } from 'app/store/models/Article';
+import type { CurrentUser, PublicUser } from 'app/store/models/User';
 import styles from './ArticleDetail.css';
+import type Config from 'config/Config';
 
 type Props = {
-  article: DetailedArticle | AdminDetailedArticle;
-  comments: Comment[];
-  loggedIn: boolean;
-  authors: DetailedUser[];
   currentUser: CurrentUser;
-  deleteComment: (id: ID, contentTarget: string) => Promise<void>;
-  emojis: Emoji[];
-  reactionsGrouped: ReactionsGrouped[];
-  fetchEmojis: () => Promise<void>;
-  fetchingEmojis: boolean;
+  loggedIn: boolean;
 };
 
-const ArticleDetail = ({
-  article,
-  authors,
-  loggedIn,
-  currentUser,
-  comments,
-  deleteComment,
-  emojis,
-  fetchEmojis,
-  fetchingEmojis,
-}: Props) => {
-  const history = useHistory();
+const ArticleDetail = ({ currentUser, loggedIn }: Props) => {
+  const { articleIdOrSlug } = useParams();
+  const article = useAppSelector((state) =>
+    selectArticleByIdOrSlug(state, { articleIdOrSlug })
+  );
+  const articleId = article && article.id;
+
+  const comments = useAppSelector((state) =>
+    selectCommentsForArticle(state, {
+      articleId,
+    })
+  );
+  const authors = useAppSelector((state) =>
+    selectUsersByIds(state, { userIds: article?.authors })
+  );
+  const emojis = useAppSelector((state) => selectEmojis(state));
+
+  const navigate = useNavigate();
   useEffect(() => {
-    history.replace(`/articles/${article.slug}`);
-  }, [article.slug, history]);
+    if (article?.slug && article?.slug !== articleIdOrSlug) {
+      navigate(`/articles/${article.slug}`, { replace: true });
+    }
+  }, [article.slug, navigate, articleIdOrSlug]);
+
+  const dispatch = useAppDispatch();
+
+  useEffect(() => {
+    if (articleIdOrSlug) {
+      dispatch(fetchArticle(articleIdOrSlug));
+    }
+  }, [articleIdOrSlug, dispatch]);
+
+  const propertyGenerator = (
+    props: { article: PublicArticle; authors: PublicUser[] },
+    config?: Partial<Config>
+  ) => {
+    if (!props.article) return;
+
+    const tags = props.article.tags.map((content) => ({
+      content,
+      property: 'article:tag',
+    }));
+
+    const authors = props.authors.map((author) => ({
+      property: 'article:authors',
+      content: `${config?.webUrl}/users/${author.username}`,
+    }));
+
+    return [
+      {
+        property: 'og:title',
+        content: props.article.title,
+      },
+      {
+        element: 'title',
+        children: props.article.title,
+      },
+      {
+        element: 'link',
+        rel: 'canonical',
+        href: `${config?.webUrl}/articles/${props.article.id}`,
+      },
+      {
+        property: 'og:type',
+        content: 'article',
+      },
+      {
+        property: 'og:image:width',
+        content: '500',
+      },
+      {
+        property: 'og:image:height',
+        content: '500',
+      },
```

Comment:
Isn't this referencing the articles cover? Wouldn't it then be wrong to say that it's a square??

Reply:
No idea - I just copied it 🤷🏼‍♂️ I'll look into it though